File: src/Java/BehavioralPatterns/State/VendingMachine/src/State.java
import java.util.List;

public abstract class State {
    protected final VendingMachine vendingMachine;

    protected State(VendingMachine vendingMachine) {
        this.vendingMachine = vendingMachine;
    }

    public abstract void insertMoney(double amount);

    public abstract void selectProduct(String productCode);

    public abstract void dispenseProduct();

    public abstract void refill(List<Product> products);

    public abstract void cancel();
}

File: src/C#/BehavioralPatterns/State/VendingMachine/State.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace VendingMachine
{
    public abstract class State
    {
        protected readonly VendingMachine vendingMachine;

        protected State(VendingMachine vendingMachine)
        {
            this.vendingMachine = vendingMachine;
        }

        public abstract void InsertMoney(decimal amount);

        public abstract void SelectProduct(string productCode);

        public abstract void DispenseProduct();

        public abstract void Refill(List<Product> products);

        public abstract void Cancel();
    }
}
